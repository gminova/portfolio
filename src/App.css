html {
  scroll-behavior: smooth;
  font-size: 26px
}

.App {
  text-align: center;
}

.menu {
  position: absolute;
  display: block;
  top: 10px;
  left: 10px;
}

.menu__button {
  position: fixed;
  background-color: #61dafb;
  color: white;
  padding: 16px;
  font-size: 16px;
  border: none;
  cursor: pointer;
  font-size: 25px;
  z-index: 1;
}

.menu__content {
  display: none;
  position: fixed;
  background-color: #f9f9f9;
  min-width: 160px;
  box-shadow: 0px 8px 16px 0px rgba(0, 0, 0, 0.2);
  z-index: 2;
}

.menu__content__link {
  padding: 12px 16px;
  display: block;
}

.menu__content__link:hover {
  background-color: #f1f1f1;
}

.menu:hover .menu__content {
  display: block;
}

.menu:hover .menu__button {
  background-color: #61dafb;
}

.section {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: #f1f1f1;
}

.profile {
  border-radius: 5px;
  max-height: 150px;
  margin-top: 10px;
}

.about__grid {
  display: grid;
  grid-template-columns: repeat(6, 1fr);
  align-content: center;
  align-items: center;
  grid-gap: 5px;
  background-color: white;
  padding: 5px;
  border-radius: 5px;
}

.logo {
  max-width: 100px;
  border: 1px solid white;
}

.logo:hover {
  opacity: 0.5;
  border: 1px solid lightgrey;
}

.menu__content__link,
.arrow__link,
.contact__link {
  color: black;
  text-decoration: none;
}

.arrow {
  background-color: #61dafb;
  text-align: center;
  font-size: 35px;
  color: f1f1f1;
  width: 50px;
  height: 50px;
  border-radius: 10%;
  animation-duration: 2s;
  animation-iteration-count: infinite;  
  margin-bottom: 60px;
}

.arrow {
  color: #f1f1f1;
  animation-name: arrow;
  animation-timing-function: linear;
}
@keyframes arrow {
  0%   { transform: translateY(0); }
  50%  { transform: translateY(-30px); }
  100% { transform: translateY(0); }
}

.section--projects {
  background-color: #f1f1f1;
  color: black;
}
.projects__gallery {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  align-content: center;
  align-items: center;
  grid-gap: 15px;
  padding: 5px;
  border-radius: 5px;
}

.projects__gallery__img {
  margin-top: 30px;
  margin-bottom: 30px;
  height: 250px;
  width: 500px;
  border-radius: 5px;
  box-shadow: 0px 8px 16px 0px rgba(0, 0, 0, 0.2);
}

.projects__gallery__img:hover {
  opacity: 0.5;
}

.contact__link {
  color: #f1f1f1;
}

.footer {
  padding: 50px;
  background-color: #61dafb;
  color: #f1f1f1;
}

@media screen and (max-width: 1025px) {

  html {
    font-size: 35px;
  }

  .projects__gallery {
    grid-template-columns: repeat(1, 1fr);
    grid-gap: 10px;
  }

  .projects__gallery__img {
    height: 350px;
    width: 700px;
  }
}

@media screen and (max-width: 900px) {

  html {
    font-size: 30px;
  }

  .about__grid {
    grid-template-columns: repeat(4, 1fr);
  }

  .projects__gallery__img {
    height: 250px;
    width: 500px;
  }
}

@media screen and (max-width: 600px) {

  html {
    font-size: 15px;
  }

  .about__grid {
    grid-template-columns: repeat(3, 1fr);
  }

  .projects__gallery__img {
    height: 150px;
    width: 300px;
  }
  
}

@media screen and (max-width: 400px) {

  .about__grid {
    grid-template-columns: repeat(2, 1fr);
  }
  
}
