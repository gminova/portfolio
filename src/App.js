import React from 'react';
import Navbar from './components/Navbar';
import About from './components/About';
import Projects from './components/Projects';
import Contact from './components/Contact';
import Footer from './components/Footer';
import './App.css';

function App() {
  return (
    <div className="App">
      <Navbar></Navbar>
      <About></About>
      <Projects></Projects>
      <Contact></Contact>
      <Footer></Footer>
    </div>
  );
}

export default App;
