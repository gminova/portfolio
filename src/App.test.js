import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';


it('renders correctly when there are no items', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});
