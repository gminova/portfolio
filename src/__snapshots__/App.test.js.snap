// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<div
  className="App"
>
  <div
    className="menu"
  >
    <button
      className="menu__button"
    >
      ☰
    </button>
    <div
      className="menu__content"
    >
      <a
        className="menu__content__link"
        href="#About"
        rel="noopener noreferrer"
      >
        About
      </a>
      <a
        className="menu__content__link"
        href="#Projects"
        rel="noopener noreferrer"
      >
        Projects
      </a>
      <a
        className="menu__content__link"
        href="#Contact"
        rel="noopener noreferrer"
      >
        Contact
      </a>
    </div>
  </div>
  <div
    className="section"
    id="About"
  >
    <img
      alt="profile"
      className="profile"
      src="https://i.imgur.com/wbshJ54.jpg"
    />
    <h1
      className="about__header"
    >
      Gergana Minova
    </h1>
    <p
      className="about__text"
    >
      I mentor prospect students at Founders and Coders. You can also find me at various events, such as London Node.js User team, TAQFull(Technology, Agility & Quality), and React meetups.
    </p>
    <p
      className="about__text"
    >
      I am committed to building software, tackling challenges and achieving goals. My objective is to join a friendly team and work towards clever and elegant solutions.
    </p>
    <div
      className="about__grid"
    >
      <img
        alt="logo"
        className="logo"
        src="https://i.imgur.com/I7jXzxR.png"
      />
      <img
        alt="logo"
        className="logo"
        src="https://i.imgur.com/3HdTez6.png"
      />
      <img
        alt="logo"
        className="logo"
        src="https://i.imgur.com/q9Gd5B2.png"
      />
      <img
        alt="logo"
        className="logo"
        src="https://i.imgur.com/ZMOBeOz.png"
      />
      <img
        alt="logo"
        className="logo"
        src="https://i.imgur.com/whQw5pf.png"
      />
      <img
        alt="logo"
        className="logo"
        src="https://i.imgur.com/xFnEsQe.png"
      />
      <img
        alt="logo"
        className="logo"
        src="https://i.imgur.com/rpMby4E.png"
      />
      <img
        alt="logo"
        className="logo"
        src="https://i.imgur.com/L48CZMZ.jpg"
      />
      <img
        alt="logo"
        className="logo"
        src="https://i.imgur.com/HDh5HyN.png"
      />
      <img
        alt="logo"
        className="logo"
        src="https://i.imgur.com/sVtEvw8.png"
      />
      <img
        alt="logo"
        className="logo"
        src="https://i.imgur.com/h30QHHZ.png"
      />
      <img
        alt="logo"
        className="logo"
        src="https://i.imgur.com/DlDXScO.png"
      />
    </div>
    <a
      className="arrow__link"
      href="#Projects"
    >
      <p
        className="arrow"
      >
        ↓
      </p>
    </a>
  </div>
  <div
    className="section section--projects"
    id="Projects"
  >
    <div
      className="projects__gallery"
    >
      <div
        className="project__card"
      >
        <a
          className="project__link"
          href="https://github.com/fac-17/Generation-Change"
          rel="noopener noreferrer"
          target="_blank"
        >
          <h3
            className="project__title"
          >
            Generation Change
          </h3>
        </a>
        <p
          className="project__text"
        >
          A serverless team project build for a client using ReactJS, SASS, BEM, Airtable, Jest, Cypress, and Netlify. This website allows educators (e.g. scout leaders) to find local social action projects for their young people.
        </p>
        <a
          className="project__link"
          href="https://github.com/fac-17/Generation-Change"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="projects__gallery__img"
            className="projects__gallery__img"
            src="https://i.imgur.com/hetklEj.jpg"
          />
        </a>
      </div>
      <div
        className="project__card"
      >
        <a
          className="project__link"
          href="https://github.com/gminova/food-miles"
          rel="noopener noreferrer"
          target="_blank"
        >
          <h3
            className="project__title"
          >
            Food Miles
          </h3>
        </a>
        <p
          className="project__text"
        >
          A full-stack team project using ExpressJS, HandlebarsJS, PostgreSQL, randomly generated dummy data, TDD, codecov, supertest, tape, Travis CI, SASS, and Heroku
        </p>
        <a
          className="project__link"
          href="https://github.com/gminova/food-miles"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="projects__gallery__img"
            className="projects__gallery__img"
            src="https://i.imgur.com/EntbhD7.png"
          />
        </a>
      </div>
      <div
        className="project__card"
      >
        <a
          className="project__link"
          href="https://github.com/gminova/latest-news"
          rel="noopener noreferrer"
          target="_blank"
        >
          <h3
            className="project__title"
          >
            Latest News
          </h3>
        </a>
        <p
          className="project__text"
        >
          A full-stack personal project using ExpressJS, HandlebarsJS, PostgreSQL, APIs, fetch, TDD, supertest, tape, authentication, JWT, Travis CI, CSS, BEM and Heroku
        </p>
        <a
          className="project__link"
          href="https://github.com/gminova/latest-news"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="projects__gallery__img"
            className="projects__gallery__img"
            src="https://github.com/gminova/latest-news/blob/master/public/images/Screenshot%20from%202019-09-05%2000-21-28.png?raw=true"
          />
        </a>
      </div>
      <div
        className="project__card"
      >
        <a
          className="project__link"
          href="https://github.com/gminova/crime-watcher"
          rel="noopener noreferrer"
          target="_blank"
        >
          <h3
            className="project__title"
          >
            Crime Watcher
          </h3>
        </a>
        <p
          className="project__text"
        >
          A team project using the xhr module, callbacks, APIs, vanilla JavaScript, TDD, leaflet maps, Heroku, HTML and CSS
        </p>
        <a
          className="project__link"
          href="https://github.com/gminova/crime-watcher"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="projects__gallery__img"
            className="projects__gallery__img"
            src="https://i.imgur.com/RLAMVR2.png"
          />
        </a>
      </div>
      <div
        className="project__card"
      >
        <a
          className="project__link"
          href="https://github.com/gminova/pearls-of-wisdom"
          rel="noopener noreferrer"
          target="_blank"
        >
          <h3
            className="project__title"
          >
            Pearls of wisdom
          </h3>
        </a>
        <p
          className="project__text"
        >
          A team project using ExpressJS, HandlebarsJS, PostgreSQL, tape, supertest, TDD, Heroku, CSS
        </p>
        <a
          className="project__link"
          href="https://github.com/gminova/pearls-of-wisdom"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="projects__gallery__img"
            className="projects__gallery__img"
            src="https://i.imgur.com/WRfEqwW.png"
          />
        </a>
      </div>
      <div
        className="project__card"
      >
        <a
          className="project__link"
          href="https://github.com/gminova/project-fear"
          rel="noopener noreferrer"
          target="_blank"
        >
          <h3
            className="project__title"
          >
            Project Fear
          </h3>
        </a>
        <p
          className="project__text"
        >
          A team project using NodeJS, http and https, TDD, API, Travis CI, tape, supertest, nock, CSS and HTML
        </p>
        <a
          className="project__link"
          href="https://github.com/gminova/project-fear"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="projects__gallery__img"
            className="projects__gallery__img"
            src="https://i.imgur.com/AOxnRcn.png"
          />
        </a>
      </div>
      <div
        className="project__card"
      >
        <a
          className="project__link"
          href="https://github.com/gminova/to-do-app"
          rel="noopener noreferrer"
          target="_blank"
        >
          <h3
            className="project__title"
          >
            To-do app
          </h3>
        </a>
        <p
          className="project__text"
        >
          A team project using HTML, CSS, vanilla JavaScript, and TDD
        </p>
        <a
          className="project__link"
          href="https://github.com/gminova/to-do-app"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="projects__gallery__img"
            className="projects__gallery__img"
            src="https://i.imgur.com/Bt2ZuXG.jpg"
          />
        </a>
      </div>
      <div
        className="project__card"
      >
        <a
          className="project__link"
          href="https://github.com/gminova/weather-app"
          rel="noopener noreferrer"
          target="_blank"
        >
          <h3
            className="project__title"
          >
            Weather app
          </h3>
        </a>
        <p
          className="project__text"
        >
          A code-along team project using ExpressJS, API and EJS
        </p>
        <a
          className="project__link"
          href="https://github.com/gminova/weather-app"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="projects__gallery__img"
            className="projects__gallery__img"
            src="https://i.imgur.com/uYnCglX.jpg"
          />
        </a>
      </div>
      <div
        className="project__card"
      >
        <a
          className="project__link"
          href="https://github.com/gminova/tic-tac-toe"
          rel="noopener noreferrer"
          target="_blank"
        >
          <h3
            className="project__title"
          >
            Tic-tac-toe
          </h3>
        </a>
        <p
          className="project__text"
        >
          A code-along personal project following the React Documentation to help me learn React syntax, classes, inheritance and states
        </p>
        <a
          className="project__link"
          href="https://github.com/gminova/tic-tac-toe"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="projects__gallery__img"
            className="projects__gallery__img"
            src="https://i.imgur.com/e0hBbEg.png"
          />
        </a>
      </div>
      <div
        className="project__card"
      >
        <a
          className="project__link"
          href="https://gminova.github.io/calc-js/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <h3
            className="project__title"
          >
            Calculator
          </h3>
        </a>
        <p
          className="project__text"
        >
          A personal project using vanilla JavaScript, HTML and CSS
        </p>
        <a
          className="project__link"
          href="https://gminova.github.io/calc-js/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="projects__gallery__img"
            className="projects__gallery__img"
            src="https://i.imgur.com/PIX7zzm.png"
          />
        </a>
      </div>
      <div
        className="project__card"
      >
        <a
          className="project__link"
          href="https://github.com/gminova/search-gifs"
          rel="noopener noreferrer"
          target="_blank"
        >
          <h3
            className="project__title"
          >
            Search GIFs
          </h3>
        </a>
        <p
          className="project__text"
        >
          A personal project using TDD, supertest, tape, TravisCI, nyc, codecov, Heroku, NodeJS and the Giphy API for searching gifs on the back-end
        </p>
        <a
          className="project__link"
          href="https://github.com/gminova/search-gifs"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="projects__gallery__img"
            className="projects__gallery__img"
            src="https://i.imgur.com/FoBXSDg.jpg"
          />
        </a>
      </div>
      <div
        className="project__card"
      >
        <a
          className="project__link"
          href="https://github.com/gminova/memory-game"
          rel="noopener noreferrer"
          target="_blank"
        >
          <h3
            className="project__title"
          >
            Japanese memory game
          </h3>
        </a>
        <p
          className="project__text"
        >
          A code-along personal project using vanilla JavaScript, HTML and CSS
        </p>
        <a
          className="project__link"
          href="https://github.com/gminova/memory-game"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="projects__gallery__img"
            className="projects__gallery__img"
            src="https://i.imgur.com/d4sSgKZ.png"
          />
        </a>
      </div>
    </div>
  </div>
  <div
    className="section"
    id="Contact"
  >
    <span
      aria-label="dart-emoji"
      role="img"
    >
      🎯
    </span>
    <a
      className="contact__link"
      href="https://twitter.com/gigi_minova"
      rel="noopener noreferrer"
      target="_blank"
    >
      Twitter
       
    </a>
    <a
      className="contact__link"
      href="https://github.com/gminova"
      rel="noopener noreferrer"
      target="_blank"
    >
      GitHub
       
    </a>
    <a
      className="contact__link"
      href="https://linkedin.com/in/gminova"
      rel="noopener noreferrer"
      target="_blank"
    >
      LinkedIn
       
    </a>
    <a
      className="contact__link"
      href="https://www.codewars.com/users/gminova"
      rel="noopener noreferrer"
      target="_blank"
    >
      CodeWars
       
    </a>
    <a
      className="contact__link"
      href="https://www.freecodecamp.org/gminova"
      rel="noopener noreferrer"
      target="_blank"
    >
      FreeCodeCamp
       
    </a>
    <a
      className="contact__link"
      download="GerganaMinovaCV.pdf"
      href="assets/GerganaMinovaCV.pdf"
    >
      Download CV.pdf
    </a>
  </div>
  <footer
    className="footer"
  >
    ©2019 Made with 
    <span
      aria-label="heart-emoji"
      role="img"
    >
      ❤️
    </span>
     by G. Minova
  </footer>
</div>
`;
